src/imex.rs

```diff
@@ -670,13 +687,19 @@ async fn export_backup(context: &Context, dir: impl AsRef<Path>) -> Result<()> {
         }
         Err(e) => {
             error!(context, "backup failed: {}", e);
-            // Not using dc_delete_file() here because it would send a DeletedBlobFile event
-            fs::remove_file(temp_path).await?;
         }
     }
 
     res
 }
+struct DeleteOnDrop(PathBuf);
+impl Drop for DeleteOnDrop {
+    fn drop(&mut self) {
+        let file = self.0.clone();
+        // Not using dc_delete_file() here because it would send a DeletedBlobFile event
+        async_std::task::spawn(async move { fs::remove_file(file).await.ok() });
```

Comment:
this should be `block_on`, to ensure proper ordering

Reply:
Done